### PR TITLE
Add linter to docs

### PR DESCRIPTION
**Pull request by m-murasovs**
Adding [markdownlint](https://github.com/DavidAnson/markdownlint) to apify-docs. 

Ended up having to add @babel/core and @babel/eslint-parser so that I could use `top-level await` examples.

Also updated ESLint to v7.

Currently, there is no severity setting for the linting, so everything that's not right comes out as an ERROR, rather than WARNING. Discussion here: https://github.com/DavidAnson/markdownlint/issues/254 and here: https://github.com/igorshubovych/markdownlint-cli/issues/75

It's not a major issue for me, since you get an email when a run fails.

With the VSCode [extension](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint), we can set it to fix many violations on Save.

### CODE REVIEW

**Comment by mtrunkat** [comment]
One minor thing - I'd separate lint and test flows in Github actions from the build and deploy. That way you will see which flows are passing and which not as you can see here in the main repo: https://cln.sh/O44meU

So you would see here:
- Code lint pass
- Markdown lint failed
- Deploy skipped (because lint failed)

**Review by mtrunkat — APPROVED** [review]

**Comment by fnesveda** [comment]
@m-murasovs The markdownlint check still fails with a bunch of issues, could you fix those as well?

**Comment by m-murasovs** [comment]
@fnesveda sorry, forgot to write - those are from the proxy docs and will be fixed when proxy docs get merged (currently there's no point fixing them). 

I've turned this into a draft until proxy docs get merged.

**Comment by m-murasovs** [comment]
Currently looking for a way to allow the 'await' keyword outside async functions, however ESLint doesn't have rules for this.

File: docs/webhooks/events.md [hunk 1]
```diff
@@ -16,19 +16,22 @@ An event is represented by a **type** and related **data**. The type is a `strin
...
 
-### Event types:
+### Event types
 
-*   `ACTOR.RUN.CREATED` - new actor run has been created.
-*   `ACTOR.RUN.SUCCEEDED` - actor run finished with status `SUCCEEDED`.
-*   `ACTOR.RUN.FAILED` - actor run finished with status `FAILED`.
-*   `ACTOR.RUN.ABORTED` - actor run finished with status `ABORTED`.
-*   `ACTOR.RUN.TIMED_OUT` - actor run finished with status `TIMED-OUT`.
+`ACTOR.RUN.CREATED` - new actor run has been created.
```
Comment (fnesveda) [inline docs/webhooks/events.md:21]: You've probably deleted the `*` here accidentally

File: docs/proxy/datacenter_proxy.md [hunk 2]
```diff
@@ -76,10 +76,17 @@ This IP/session ID combination persists and expires 24 hours later. Each additio
 
 So, if you use the session at least once a day, it will never expire, with two possible exceptions:
 
-*   The proxy server stops responding and is marked as [dead](#dead-proxies) during a health check.
-*   If the proxy server is part of a proxy group that is refreshed monthly and is rotated out.
+* The proxy server stops responding and is marked as [dead](#dead-proxies) during a health check.
+* If the proxy server is part of a proxy group that is refreshed monthly and is rotated out.
 
+<<<<<<< HEAD
```
Comment (fnesveda) [inline docs/proxy/datacenter_proxy.md:82]: You missed a merge conflict here.

**Review by fnesveda — CHANGES_REQUESTED** [review]
There's a lot of instances where there's a code block containing just an URL or just one line of text, but the code language is set as `json`. I'd use `text` there, it's gonna be less confusing when rendered in the code tabs.

File: docs/actors/development/source_code.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ Similarly as with the [Git repository]({{@link actors/development/source_code.md
 
 Internally, Apify uses Docker to build and run actors. To control the build of the actor, you can create a custom **Dockerfile** in the root of the Git repository or Zip directory. Note that this option is not available for the [Single JavaScript file]({{@link actors/development/source_code.md#single-javascript-file}}) option. If the **Dockerfile** is missing, the system uses the following default:
 
-```json
+```text
```
Comment (fnesveda) [inline docs/actors/development/source_code.md:85]: This could actually be `dockerfile` instead of `text`

**Review by fnesveda — CHANGES_REQUESTED** [review]

**Review by fnesveda — APPROVED** [review]

**Review by fnesveda — APPROVED** [review]

**Review by mtrunkat — APPROVED** [review]
Looks good!

**Pull request merged**